docs/source/DeveloperGuide/Systems/Transformation/bodyplugin.rst

```diff
@@ -0,0 +1,131 @@
+
+
+The ReplicateOrMove Plugin Example
+----------------------------------
+
+Imagine a world where you would have storages that are reliable, and some that are less. And imagine that you are creating a transformation to shuffle data around. Now, if the destination of your file is a trustworthy one, you could imagine deleting the original copy. But if the destination is likely to lose your file, you are better of keeping a second copy. That's what our example plugin will do.
+
+.. code-block:: python
+   :caption: ReplicateOrMoveBody.py
+   :linenos:
+
+
+    from DIRAC import gLogger
+    from DIRAC.RequestManagementSystem.Client.Request import Request
+    from DIRAC.RequestManagementSystem.Client.Operation import Operation
+    from DIRAC.RequestManagementSystem.Client.File import File
+
+    from DIRAC.TransformationSystem.Client.BodyPlugin.BaseBody import BaseBody
+
+
+    sLog = gLogger.getLocalSubLogger(__name__)
+
+
+    class ReplicateOrMoveBody(BaseBody):
+        """
+        If the destination storage is trustworthy, move the file there,
+        otherwise, just do a copy of the file
+        """
+
+        # This is needed to know how to serialize the object
+        _attrToSerialize = ["trustworthy", "undependable"]
+
+        def __init__(self, trustworthy=None, undependable=None):
+            """C'tor
+            :param list trustworthy: trustworthy storage to which we will move the file
+            :param list undependable: doubtful storage to which we will rather just copy
+            """
+            self.trustworthy = trustworthy
+            self.undependable = undependable
+
+        def taskToRequest(self, taskID, task, transID):
+            """
+            Convert a task into a Request with either a ReplicateAndRegister
+            or a Move Operation
+            """
+            req = Request()
+            op = Operation()
+
+            # Decide on an operation type
+            targetSE = task["TargetSE"]
+            if targetSE in self.trustworthy:
+                op.Type = "MoveReplica"
+            elif targetSE in self.undependable:
+                op.Type = "ReplicateAndRegister"
+            else:
+                sLog.warn("Hum, SE is not in a known list... let's try to assess the quality of the storage...")
+                # VERY doubtful storage
+                if "RAL" in targetSE:
+                    op.Type = "ReplicateAndRegister"
+                else:
+                    op.Type = "MoveReplica"
+
+            for lfn in task["InputData"]:
+                trFile = File()
+                trFile.LFN = lfn
+
+                op.addFile(trFile)
+
+            req.addOperation(op)
+
+            return req
+
+
+
+Using the Body Plugin
+---------------------
+
+When a transformation is created, set the `EvenOdd` plugin with `setPlugin` and set the 'EvenOdd' parameter to 'Odd'
+with `setEvenOdd`, and then execute this function to test it.
```

Comment:
What's this phrase? Sorry I don't understand it.

Reply:
oh sorry that's a copy paste from the other example, which does not make much sense here